File: app/src/main/java/com/mkrworld/mobilpay/dto/mobilenumberstatus/DTOMobileNumberStatusResponse.java
package com.mkrworld.mobilpay.dto.mobilenumberstatus;

import com.google.gson.annotations.SerializedName;
import com.mkrworld.androidlib.utils.Tracer;
import com.mkrworld.mobilpay.BuildConfig;

/**
 * Created by mkr on 27/3/18.
 */

public class DTOMobileNumberStatusResponse {
    private static final String TAG = BuildConfig.BASE_TAG + ".DTOMobileNumberStatusResponse";

    @SerializedName("message")
    private String mMessage;

    @SerializedName("data")
    private Data mData;

    /**
     * Method to get the API Message
     *
     * @return
     */
    public String getMessage() {
        return mMessage;
    }

    /**
     * Method to get the API Data
     *
     * @return
     */
    public Data getData() {
        return mData;
    }

    /**
     * Class to hold the Response Data
     */
    public class Data {

        @SerializedName("status")
        private String mStatus;

        /**
         * Method to get Status
         *
         * @return
         */
        public String getStatus() {
            Tracer.debug(TAG, "getStatus : ");
            return mStatus != null ? mStatus : "0";
        }
    }
}

File: app/src/main/java/com/mkrworld/mobilpay/dto/mobilenumberstatus/DTOMobileNumberStatusResponse.kt
package com.mkrworld.mobilpay.dto.mobilenumberstatus

import android.content.ContentValues.TAG
import com.google.gson.annotations.SerializedName
import com.mkrworld.androidlib.utils.Tracer
import com.mkrworld.mobilpay.BuildConfig

/**
 * Created by mkr on 27/3/18.
 */

class DTOMobileNumberStatusResponse {

    companion object {
        private val TAG = BuildConfig.BASE_TAG + ".DTOMobileNumberStatusResponse"
    }

    @SerializedName("message")
    private var mMessage : String? = null

    @SerializedName("mData")
    private var mData : Data? = null

    /**
     * Method to get the API Data
     *
     * @return
     */
    fun getData() : Data {
        return mData!!
    }

    /**
     * Method to get the API Message
     *
     * @return
     */
    fun getMessage() : String {
        return mMessage ?: ""
    }

    /**
     * Class to hold the Response Data
     */
    inner class Data {

        @SerializedName("status")
        private val mStatus : String? = null

        /**
         * Method to get Status
         *
         * @return
         */
        val status : String
            get() {
                Tracer.debug(TAG, "getStatus : ")
                return mStatus ?: "0"
            }
    }
}
